File: Python/225__Implement_Stack_using_Queues.py
class MyStack(object):

    def __init__(self):
        """
        Initialize your data structure here.
        """
        self.data = []


    def push(self, x):
        """
        Push element x onto stack.
        :type x: int
        :rtype: void
        """
        self.data = [x] + self.data


    def pop(self):
        """
        Removes the element on top of the stack and returns that element.
        :rtype: int
        """
        x = self.data[0]
        self.data = self.data[1:]
        return x


    def top(self):
        """
        Get the top element.
        :rtype: int
        """
        return self.data[0]


    def empty(self):
        """
        Returns whether the stack is empty.
        :rtype: bool
        """
        return len(self.data) == 0


# Your MyStack object will be instantiated and called as such:
# obj = MyStack()
# obj.push(x)
# param_2 = obj.pop()
# param_3 = obj.top()
# param_4 = obj.empty()

File: Java/225__Implement_Stack_using_Queues.java
class MyStack {

    Queue<Integer> queue;

    /** Initialize your data structure here. */
    public MyStack() {
        queue = new LinkedList<>();
    }

    /** Push element x onto stack. */
    public void push(int x) {
        queue.offer(x);
    }

    /** Removes the element on top of the stack and returns that element. */
    public int pop() {
        int limit = queue.size() - 1, index = 0;
        while (index != limit) {
            queue.offer(queue.poll());
            index++;
        }
        return queue.poll();
    }

    /** Get the top element. */
    public int top() {
        int limit = queue.size(), index = 0;
        int result = 0;
        while (index != limit) {
            result = queue.poll();
            queue.offer(result);
            index++;
        }
        return result;
    }

    /** Returns whether the stack is empty. */
    public boolean empty() {
        return queue.isEmpty();
    }
}

/**
 * Your MyStack object will be instantiated and called as such:
 * MyStack obj = new MyStack();
 * obj.push(x);
 * int param_2 = obj.pop();
 * int param_3 = obj.top();
 * boolean param_4 = obj.empty();
 */
